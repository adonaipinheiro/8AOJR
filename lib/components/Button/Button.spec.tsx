import React from "react";
import "@testing-library/jest-dom";
import { render, screen, fireEvent } from "@testing-library/react";
import { describe, it, expect, vi } from "vitest";
import { Button, ButtonProps } from ".";

const renderButton = (props?: Partial<ButtonProps>) => {
  return render(<Button text="Entrar" buttonStyle="primary" {...props} />);
};

describe("@components/Button", () => {
  it("should render the button with the correct text", () => {
    renderButton();
    expect(screen.getByRole("button", { name: "Entrar" })).toBeInTheDocument();
  });

  it("should render with left and right icons", () => {
    renderButton({
      leftIcon: <span data-testid="left-icon">L</span>,
      rightIcon: <span data-testid="right-icon">R</span>,
    });
    expect(screen.getByTestId("left-icon")).toBeInTheDocument();
    expect(screen.getByTestId("right-icon")).toBeInTheDocument();
  });

  it("should call onClick handler when clicked", () => {
    const onClick = vi.fn();
    renderButton({ onClick });
    fireEvent.click(screen.getByRole("button"));
    expect(onClick).toHaveBeenCalledTimes(1);
  });

  it("should be disabled when the disabled prop is set", () => {
    renderButton({ disabled: true });
    const button = screen.getByRole("button");
    expect(button).toBeDisabled();
  });

  it("should have correct aria attributes", () => {
    renderButton({ disabled: true });
    const button = screen.getByRole("button");
    expect(button).toHaveAttribute("aria-disabled", "true");
  });

  it("should match to snapshot", () => {
    const { asFragment } = renderButton();
    expect(asFragment()).toMatchSnapshot();
  });
});
