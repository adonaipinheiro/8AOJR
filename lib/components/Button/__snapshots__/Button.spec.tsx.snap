// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`@components/Button > should match to snapshot 1`] = `
<DocumentFragment>
  <button
    aria-label="Entrar"
    buttonstyle="primary"
    class="sc-gTrWKq dbtOAl"
  >
    <p
      class="sc-dDvxFM coBwAw"
    >
      Entrar
    </p>
  </button>
</DocumentFragment>
`;
